### DISCUSSION TITLE
Posteriors for random intercepts

### DISCUSSION THREAD

**Discussion by stefgehrig** [liesel-devs/liesel#182] [Q&A] [answered]
Is there a simple and efficient way to generate draws for the random intercepts for all units in the scale and location parameter models in a model like this?

`{rliesel}` code for model:

```r
model <- liesel(
  response = df$y,
  distribution = "LogNormal",
  predictors = list(
    loc = predictor( ~ s(bmi, bs = "ps", k = 15) + s(center, bs = "re"), inverse_link = "Identity"),
    scale = predictor( ~ s(center, bs = "re"), inverse_link = "Exp")
  ),
  data = df
)
```

**Comment by jobrachem** [comment, marked as answer]
I would expect the samples to be available directly. Can you show me the output of the following code?

```python
samples = results.get_posterior_samples()
list(samples)
```

Here, `results` is the object returned by [`Engine.get_results()`](https://docs.liesel-project.org/en/latest/generated/liesel.goose.Engine.get_results.html#liesel.goose.Engine.get_results). The snippet above will output a list of the names of parameters for which we have drawn posterior samples.

**Reply by stefgehrig** [reply]
You are absolutely right. They are in the respective `_beta` suffix variables. Which is where they should be. Thanks a lot
